mempool/src/pool/feerate_points.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::collections::BTreeMap;
+
+use common::primitives::Amount;
+
+use super::fee::Fee;
+
+pub fn linear_interpolation(x0: usize, y0: u128, x1: usize, y1: u128, x: usize) -> Option<u128> {
```

Comment:
It's much more meaningful to make this tuples (x0, y0), (x1, y1), and x alone. 

Also, it would be really nice to make this work with generic types X and Y so that it can be reused/moved if necessary. I think it would be easy to do. The trait bounds are simply some math ops that will be given to you by the compiler, quite similar to the ones I did with the timing resistant equality in the crypto library. If it's too much work, don't do it.

Reply:
fixed to use generics